homeassistant/components/broadlink/sensor.py

```diff
@@ -35,28 +44,40 @@
     "noise": ["Noise", " "],
 }
 
+DEVICE_TYPES = A1_TYPES + RM_TYPES + RM4_TYPES
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
-        vol.Optional(CONF_NAME, default=DEVICE_DEFAULT_NAME): vol.Coerce(str),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): vol.Coerce(str),
         vol.Optional(CONF_MONITORED_CONDITIONS, default=[]): vol.All(
             cv.ensure_list, [vol.In(SENSOR_TYPES)]
         ),
-        vol.Required(CONF_HOST): cv.string,
-        vol.Required(CONF_MAC): cv.string,
+        vol.Required(CONF_HOST): vol.All(vol.Any(hostname, ip_address), cv.string),
+        vol.Required(CONF_MAC): mac_address,
+        vol.Optional(CONF_TYPE, default=DEVICE_TYPES[0]): vol.In(DEVICE_TYPES),
         vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
     }
 )
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Broadlink device sensors."""
-    host = config.get(CONF_HOST)
-    mac = config.get(CONF_MAC).encode().replace(b":", b"")
-    mac_addr = binascii.unhexlify(mac)
-    name = config.get(CONF_NAME)
-    timeout = config.get(CONF_TIMEOUT)
+    host = config[CONF_HOST]
+    mac_addr = config[CONF_MAC]
+    model = config[CONF_TYPE]
+    name = config[CONF_NAME]
+    timeout = config[CONF_TIMEOUT]
     update_interval = config.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL)
-    broadlink_data = BroadlinkData(update_interval, host, mac_addr, timeout)
+
+    if model in RM4_TYPES:
+        api = blk.rm4((host, DEFAULT_PORT), mac_addr, None)
+        check_sensors = api.check_sensors
+    else:
+        api = blk.a1((host, DEFAULT_PORT), mac_addr, None)
+        check_sensors = api.check_sensors_raw
```

Comment:
This is working, I tested it on my RM Pro. So it necessarily has to work for the RM4 class, which has the same interface. What I am asking for here is not even a leap of faith, but a simple logical deduction. If it works for one, it has to work for the other.